src/soap/handlers/GetSiteInfoObject.js

```diff
@@ -68,9 +68,8 @@ export async function* getSiteInfoObject(
     : undefined
 
   // Fetch stations
-  const stations = await helpers.findOne(
+  const stations = await helpers.findMany(
```

Comment:
This part is going to hurt, but I believe it will match the current behavior (somewhat).

I prefer to use findOne for a single station, but in this case we don't have the id. So we'll use findMany with a limit. Also note that the slug is required. This pattern will match the GetSites pattern for some consistency and sanity. If the parts are missing, then it will short circuit the query with values that are just dashes (kinda gross but it works). Then check the stations.length since retuning a station is required apparently. I got a SOAP fault when my site code was wrong. The real SOAP service returned "Object reference not set to an instance of an object.". Weird. I think we can return "Station not found" for now (makes more sense).

```
  // Fetch stations
  const stations = await helpers.findMany(
    'stations',
    Object.assign(
      {
        is_enabled: true,
        is_hidden: false,
        $limit: 1
      },
      organization &&
        organization.data &&
        organization.data.length &&
        organization.data[0]._id
        ? { organization_id: organization.data[0]._id }
        : undefined,
      {
        slug: (org || parts[0] || '-') + '-' + (parts[1] || '-')
      }
    )
  )

  if (!stations.length) throw new Error('Station not found')
```

Further down, I would just add a comment.

```
  // Will only be one station since $limit=1
  for (const station of stations) {
```

This is the behavior I saw. It's only ONE station is returned EVER. If the site code is invalid, then an error is returned. Please double check.